File: euler_073.py
from time import time
from math import gcd

def next_f(f1, f2):
    [p, pm] = [f1[0], f2[0]]
    [q, qm] = [f1[1], f2[1]]
    d = int((q + limit) / qm)
    pmm = d * pm - p
    qmm = d * qm - q
    g = gcd(pmm, qmm)
    return (pmm // g, qmm // g)


start = time()

limit = 12000

farey = [(1, limit), (1, limit - 1)]

f = next_f(farey[-2], farey[-1])

while f[0] != f[1]:
    farey.append(f)
    f = next_f(farey[-2], farey[-1])

third = farey.index((1, 3))
half = farey.index((1, 2))

print("Number of fractions: " + str(half - third - 1))

stop = time()
print("Time: " + str(stop - start))

File: euler_073.js
const start = Date.now();

const isLarger = (f1, f2) => {
    return f1[0] * f2[1] > f1[1] * f2[0];
}

const gcd = (a, b) => {
  if (b > a) {
    return gcd(b, a);
  }
  while (true) {
    a %= b;
    if (a === 0) {
      return b;
    }
    b %= a;
    if (b === 0) {
      return a;
    }
  }
};

const lower = [1,3];
const upper = [1,2];

let fractions = new Set();

for (let d = 2; d < 12001; d++) {
    let nums = [...Array(d).keys()].slice(1).filter(x => gcd(d, x) === 1);
    for (const n of nums) {
        if (isLarger([n, d], lower)  && isLarger(upper, [n,d])) {
            fractions.add([n, d]);
        }
    }
}

const res = fractions.size;


const end = Date.now();

console.log("Execute time: ", end - start, " The number of fractions is: ", res);